File: TaipeiParkC/TaipeiParkC/ParkDetailViewController.m
//
//  ParkDetailViewController.m
//  TaipeiParkC
//
//  Created by Richard on 2018/02/06.
//  Copyright © 2018年 Snowowl. All rights reserved.
//

#import "ParkDetailViewController.h"
#import "ParkCollectionViewCell.h"

@interface ParkDetailViewController ()

@end

@implementation ParkDetailViewController

@synthesize parkItem;
@synthesize nearbyviewArray;

- (void)setAScrollView:(UIScrollView *)aScrollView
{
    _aScrollView = aScrollView;
    self.aScrollView.scrollEnabled = YES;
    self.aScrollView.showsVerticalScrollIndicator = YES;
    self.aScrollView.showsHorizontalScrollIndicator = NO;
    
}

- (void)setACollectionView:(UICollectionView *)aCollectionView
{
    _aCollectionView = aCollectionView;
    self.aCollectionView.backgroundColor = [UIColor whiteColor];
    self.aCollectionView.dataSource = self;
    self.aCollectionView.delegate = self;
    self.aCollectionView.scrollEnabled = YES;
    self.aCollectionView.showsHorizontalScrollIndicator = YES;
    self.aCollectionView.showsVerticalScrollIndicator = NO;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    [self loadData];
}

- (void)loadData
{
    if (parkItem) {
        //        NSString *imageURL = [[parkItem objectForKey:@"Image"] description];
        NSString *parkName = [[parkItem objectForKey:@"ParkName"] description];
        NSString *name = [[parkItem objectForKey:@"Name"] description];
        NSString *introduction = [[parkItem objectForKey:@"Introduction"] description];
        NSString *openTime = [[parkItem objectForKey:@"OpenTime"] description];
        //        NSString *yearBuilt = [[parkItem objectForKey:@"YearBuilt"] description];
        
        UIImage *image = (UIImage*)[parkItem objectForKey:@"thumbImage"];
        
        self.parkNameLabel.text = parkName;
        self.nameLabel.text = name;
        self.openTimeLabel.text = openTime;
        self.parkIntroLabel.text = introduction;
        self.parkImageView.image = image;
        
        [self.parkIntroLabel sizeToFit];
        
        int nScrollContentSize = self.parkImageView.frame.size.height + self.parkNameLabel.frame.size.height + self.nameLabel.frame.size.height + self.openTimeLabel.frame.size.height + self.parkIntroLabel.frame.size.height + self.aCollectionView.frame.size.height;
        nScrollContentSize += 50;
        
        CGRect contentViewFrame = CGRectMake(0, 0, self.aScrollView.frame.size.width, nScrollContentSize);
        self.scrollContentView.frame = contentViewFrame;
        
        self.aScrollView.contentSize = CGSizeMake(contentViewFrame.size.width, contentViewFrame.size.height);
    }
}

#pragma mark <UICollectionViewDataSource>

- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView
{
    return 1;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    if (nearbyviewArray.lastObject) {
        return nearbyviewArray.count;
    } else {
        return 0;
    }
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    ParkCollectionViewCell *cell = (ParkCollectionViewCell*)[collectionView dequeueReusableCellWithReuseIdentifier:@"PhotoCell" forIndexPath:indexPath];

    NSDictionary *cellItem = [nearbyviewArray objectAtIndex:indexPath.row];
    
    UIImage *image = (UIImage*)[cellItem objectForKey:@"thumbImage"];
    cell.photoImageView.image = image;
    cell.nameLabel.text = [[cellItem objectForKey:@"Name"] description];
    return cell;
}

#pragma mark <UICollectionViewDelegate>

- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath
{
    NSDictionary *cellItem = [nearbyviewArray objectAtIndex:indexPath.row];

    parkItem = cellItem;
    [self loadData];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: TaiepiParkS/TaiepiParkS/ParkDetailViewController.swift
//
//  ParkDetailViewController.swift
//  TaiepiParkS
//
//  Created by Richard on 2018/02/22.
//  Copyright © 2018年 Snowowl. All rights reserved.
//

import UIKit

class ParkDetailViewController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegate {

    @IBOutlet weak var aScrollView: UIScrollView! {
        didSet {
            aScrollView.isScrollEnabled = true
            aScrollView.showsVerticalScrollIndicator = true
            aScrollView.showsHorizontalScrollIndicator = true
        }
    }
    
    @IBOutlet weak var scrollContentView: UIView!
    
    @IBOutlet weak var parkImageView: UIImageView!
    
    @IBOutlet weak var parkNameLabel: UILabel!
    
    @IBOutlet weak var nameLabel: UILabel!
    
    @IBOutlet weak var openTimeLabel: UILabel!
    
    @IBOutlet weak var parkIntroLabel: UILabel!
    
    @IBOutlet weak var aCollection: UICollectionView! {
        didSet {
            aCollection.backgroundColor = UIColor.white
            aCollection.dataSource = self
            aCollection.delegate = self
            aCollection.isScrollEnabled = true
            aCollection.showsVerticalScrollIndicator = false
            aCollection.showsHorizontalScrollIndicator = true
        }
    }
    
    var parkItem : NSDictionary?
    var nearbyviewArray : NSArray?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        loadData()
    }
    
    func loadData() {
        if parkItem != nil {
            let parkName = parkItem?.object(forKey: "ParkName") as! String
            let name = parkItem?.object(forKey: "Name") as! String
            let introduction = parkItem?.object(forKey: "Introduction") as! String
            let openTime = parkItem?.object(forKey: "OpenTime") as! String
            
            let image = parkItem?.object(forKey: "thumbImage") as! UIImage
            
            self.parkNameLabel.text = parkName
            self.nameLabel.text = name
            self.openTimeLabel.text = openTime
            self.parkIntroLabel.text = introduction
            self.parkImageView.image = image
            
            self.parkIntroLabel.sizeToFit()
            
            var scrollContentSize = self.parkImageView.frame.size.height + self.parkNameLabel.frame.size.height + self.nameLabel.frame.size.height + self.openTimeLabel.frame.size.height + self.parkIntroLabel.frame.size.height + self.aCollection.frame.size.height
            scrollContentSize += 50
            
            let contentViewFrame = CGRect(x: 0, y: 0, width: self.aScrollView.frame.size.width, height: scrollContentSize)
            
            self.scrollContentView.frame = contentViewFrame
            
            self.aScrollView.contentSize = CGSize(width: contentViewFrame.size.width, height: contentViewFrame.size.height)
        }
    }

    // MARK: UICollectionViewDataSource
    
    func numberOfSections(in collectionView: UICollectionView) -> Int {
        return 1
    }
    
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        if nearbyviewArray?.lastObject != nil {
            return nearbyviewArray!.count
        } else {
            return 0
        }
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "PhotoCell", for: indexPath) as? ParkCollectionViewCell
        let cellItem = nearbyviewArray![indexPath.row] as! NSDictionary
        
        let image = cellItem.object(forKey: "thumbImage") as! UIImage
        cell!.photoImageView.image = image
        cell!.nameLabel.text = cellItem.object(forKey: "Name") as! String?
        return cell!
    }
    
    // MARK: UICollectionViewDelegate
    
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
        let cellItem = nearbyviewArray![indexPath.row] as! NSDictionary
        
        parkItem = cellItem
        loadData()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
